.github/workflows/ci-esp.yml

```diff
@@ -0,0 +1,46 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+  schedule:
+    - cron: '50 4 * * *'
+
+env:
+  rust_toolchain: nightly-2022-04-07
+  target: xtensa-esp32-espidf
+
+jobs:
+  compile:
+    name: Compile
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup | Checkout
+        uses: actions/checkout@v2
+      - name: Setup | Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: ${{ env.rust_toolchain }}
+          components: rustfmt, clippy
+      - name: Setup | Dependencies
+        run: sudo apt install -y git curl gcc ninja-build cmake libudev-dev python3 python3-pip libusb-1.0-0 libssl-dev pkg-config libtinfo5 libclang1 libclang-dev
+      - name: Setup | Rust Xtensa toolchain
+        run: curl -LO https://raw.githubusercontent.com/esp-rs/rust-build/main/install-rust-toolchain.sh; chmod +x install-rust-toolchain.sh; ./install-rust-toolchain.sh --export-file ~/.xtensa_exports
```

Comment:
Might be preferable to use https://github.com/esp-rs/xtensa-toolchain GHA, should hopefully make things simpler. See this other PR on esp-idf-sys: https://github.com/esp-rs/esp-idf-sys/pull/102

Reply:
Oh thank you